[CI] Add jazzy :rocket:

christophfroehlich opened the pull request:
- Jazzy jobs + readme
- because all of the repos file are empty, there was no difference between binary and semi-binary. So I deleted also the semi-binary workflow file.

Jobs for jazzy-main are failing, but jazzy-testing works fine already.

saikishor commented on realtime_tools-not-released.jazzy.repos:
```diff
@@ -0,0 +1,6 @@
+repositories:
+  ## EXAMPLE DEPENDENCY
+#   <some_ros_package>:
+#     type: git
+#     url: git@github.com:<some_github_namespace>/<some_ros_package>.git
+#     version: master
```
@christophfroehlich do we need these files?

saikishor commented on realtime_tools.jazzy.repos:
```diff
@@ -0,0 +1,6 @@
+repositories:
+  ## EXAMPLE DEPENDENCY
+#   <some_ros_package>:
+#     type: git
+#     url: git@github.com:<some_github_namespace>/<some_ros_package>.git
+#     version: master
```
This one too?

saikishor reviewed (commented):
Rest looks good to me :)

christophfroehlich commented:
@saikishor I haven't questioned this, just added jazzy. But you are right, it works without specifying a repos file. And because all of them are empty, there was no difference between binary and semi-binary. So I deleted also the semi-binary workflow file.

christophfroehlich commented on .github/workflows/ci-coverage-build.yml:
```diff
@@ -11,7 +11,7 @@ on:
   coverage:
     name: coverage build
-    uses: ros-controls/ros2_control_ci/.github/workflows/reusable-build-coverage.yml@master
+    uses: ros-controls/ros2_control_ci/.github/workflows/reusable-build-coverage.yml@optional_repos
```
```suggestion
    uses: ros-controls/ros2_control_ci/.github/workflows/reusable-build-coverage.yml@master
```

christophfroehlich replied:
once https://github.com/ros-controls/ros2_control_ci/pull/64 is merged

saikishor replied:
Awesome :+1:

saikishor reviewed (approved):
Looking good to me. Thanks for the changes :)

christophfroehlich commented on .github/workflows/ci-coverage-build.yml:
```diff
@@ -11,8 +11,7 @@ on:
   coverage:
     name: coverage build
-    uses: ros-controls/ros2_control_ci/.github/workflows/reusable-build-coverage.yml@master
+    uses: ros-controls/ros2_control_ci/.github/workflows/reusable-build-coverage.yml@coverage_build
```
```suggestion
    uses: ros-controls/ros2_control_ci/.github/workflows/reusable-build-coverage.yml@master
```

christophfroehlich commented on .github/workflows/binary-build.yml:
```diff
@@ -20,10 +20,9 @@ jobs:
       fail-fast: false
       matrix:
-        ROS_DISTRO: [rolling, iron, humble]
+        ROS_DISTRO: [rolling, jazzy, iron, humble]
```
```suggestion
        ROS_DISTRO: [humble, iron, jazzy, rolling]
```

christophfroehlich commented on .github/workflows/debian-build.yml:
```diff
@@ -15,8 +15,7 @@ jobs:
       fail-fast: false
       matrix:
-        ROS_DISTRO: [rolling, iron, humble]
+        ROS_DISTRO: [rolling, iron, jazzy, humble]
```
```suggestion
        ROS_DISTRO: [humble, iron, jazzy, rolling]
```

christophfroehlich commented on .github/workflows/rhel-semi-binary-build.yml:
```diff
@@ -15,8 +15,7 @@ jobs:
       fail-fast: false
       matrix:
-        ROS_DISTRO: [rolling, iron, humble]
+        ROS_DISTRO: [rolling, iron, jazzy, humble]
```
```suggestion
        ROS_DISTRO: [humble, iron, jazzy, rolling]
```
